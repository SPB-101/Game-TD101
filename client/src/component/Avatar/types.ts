import React from "react";

export type Props = React.ImgHTMLAttributes<HTMLImageElement>;
