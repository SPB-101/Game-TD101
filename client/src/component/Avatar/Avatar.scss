.avatar {
  box-shadow: var(--shadow-purple);
}
