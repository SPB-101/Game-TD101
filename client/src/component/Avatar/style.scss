.avatar {
  object-fit: cover;
  border: var(--border) var(--secondary-border-color);
}
