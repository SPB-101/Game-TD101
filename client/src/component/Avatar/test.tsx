import React from "react";
import { shallow } from "enzyme";
import { Avatar } from "./index";
import ProfileDefaultIcon from "../../assets/images/icons/profile-icon.svg";

describe("<Avatar />", () => {
  it("should render correctly without props", () => {
    const wrapper = shallow(<Avatar />);
    expect(wrapper.matchesElement(<ProfileDefaultIcon />)).toBe(true);
  });

  it("should have a default class 'avatar'", () => {
    const wrapper = shallow(<Avatar />);
    expect(wrapper.hasClass("avatar")).toBe(true);
  });

  it("should render correctly with props", () => {
    const src = "./john.png";
    const wrapper = shallow(<Avatar src={src} />);
    expect(wrapper.props().src).toEqual(src);
  });
});
