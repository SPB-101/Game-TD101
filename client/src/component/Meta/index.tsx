import React from "react";
import { Helmet } from "react-helmet-async";

import { data } from "./seo";

export const Meta = () => {
  return (
    <Helmet>
      <title>{data.title}</title>

      <meta name="description" content={data.description} />
      <meta property="og:type" content="website" />
      <meta property="og:title" content={data.title} />
      <meta property="og:description" content={data.description} />
      <meta property="og:image" content={data.image} />
      <meta property="og:site_name" content="Game TD101" />
      <meta name="title" content={data.title} />

      <meta
        name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"
      />
    </Helmet>
  );
};
