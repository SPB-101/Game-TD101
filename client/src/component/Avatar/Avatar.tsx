import React, { SyntheticEvent } from "react";
import classNames from "classnames";

import "./Avatar.scss";
import ProfileDefaultIcon from "../../assets/images/icons/profile-icon.svg";
import ImageErrorIcon from "../../assets/images/icons/attachment-error-icon.png";

import { Props } from "./types";

export const Avatar = ({
  className,
  src,
  alt,
  width = 100,
  height = 100,
}: Props) => {
  const errorHandler = (e: SyntheticEvent<HTMLImageElement>) => {
    e.currentTarget.onerror = null;
    e.currentTarget.src = ImageErrorIcon;
  };

  const avatarClasses = classNames("avatar", className);

  const w = `${width}px`;
  const h = `${height}px`;

  if (!src) {
    return (
      <ProfileDefaultIcon width={w} height={h} className={avatarClasses} />
    );
  }

  return (
    <img
      className={avatarClasses}
      src={src}
      alt={alt}
      width={w}
      height={h}
      onError={(e) => errorHandler(e)}
    />
  );
};
