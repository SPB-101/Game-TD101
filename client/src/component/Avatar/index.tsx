import React, { SyntheticEvent } from "react";
import classNames from "classnames";

import { Props } from "./types";

import "./style.scss";
import ImageErrorIcon from "@assets/images/icons/attachment-error-icon.png";
import ProfileDefaultIcon from "@assets/images/icons/profile-icon.svg";

export const Avatar = ({
  className,
  src,
  alt,
  width = 100,
  height = 100,
}: Props) => {
  const errorHandler = (e: SyntheticEvent<HTMLImageElement>) => {
    e.currentTarget.onerror = null;
    e.currentTarget.src = ImageErrorIcon;
  };

  const avatarClasses = classNames("avatar", className);

  const w = `${width}px`;
  const h = `${height}px`;

  if (!src) {
    return (
      <ProfileDefaultIcon className={avatarClasses} width={w} height={h} />
    );
  }

  return (
    <img
      className={avatarClasses}
      src={src}
      alt={alt}
      width={w}
      height={h}
      onError={(e) => errorHandler(e)}
    />
  );
};
