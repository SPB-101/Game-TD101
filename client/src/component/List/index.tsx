import React from "react";
import { useTranslation } from "react-i18next";
import classNames from "classnames";

import { Loader } from "@component/Loader";

import type { Props } from "./types";

import "./style.scss";

export const List = ({
  className,
  isLoading,
  count,
  emptyText,
  children,
}: Props) => {
  const { t } = useTranslation();
  const listClass = classNames("list", { [`${className}`]: className });
  const getEmptyText = emptyText ? emptyText : t("emptyList");
  const isEmpty = count === 0;

  if (isLoading) {
    return <Loader />;
  }

  return <ul className={listClass}>{isEmpty ? getEmptyText : children}</ul>;
};
