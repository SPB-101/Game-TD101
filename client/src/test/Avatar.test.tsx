import React from "react";
import renderer from "react-test-renderer";
import { Avatar } from "../component/Avatar";

describe("<Avatar />", () => {
  it("Avatar renders correctly", () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
