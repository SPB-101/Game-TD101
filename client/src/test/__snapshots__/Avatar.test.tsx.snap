// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Avatar renders correctly 1`] = `
<profile-icon.svg
  className="avatar"
  height="50px"
  width="50px"
/>
`;
